addons/mass_mailing_sms/models/sms_tracker.py

```diff
@@ -47,9 +47,11 @@ def _update_sms_traces(self, trace_status, failure_type=False, failure_reason=Fa
                 'trace_status': trace_status,
                 'failure_type': failure_type,
                 'failure_reason': failure_reason,
-                **{'sent_datetime': fields.Datetime.now() if trace_status == 'pending' else {}},
             }
             traces.write(traces_values)
+            traces.filtered(
+                lambda t: t.trace_status not in ['outgoing', 'process', 'error', 'cancel'] and not t.sent_datetime
+            ).sent_datetime = fields.Datetime.now()
```

Comment:
Could you use ``self.env.cr.now`` ? That way it is always set at transaction datetime, for technical records it is better as it makes datetimes more deterministic (notably for tests ^^ ).

Reply:
Hii, @tde-banana-odoo :wave: 

I don't think we currently have a way to directly get `delivered` status in test cases, so I slightly modify the `mockSMSGateway` to get that status for my test case.